event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventJdbcRepository.java

```diff
@@ -198,14 +226,81 @@ protected String jndiName() throws NamingException {
     }
 
     @Override
-    public Stream<Event> getPage(final long offset, final long pageSize,  final Map<String, Object> params) {
+    public Stream<Event> getFeed(final long offset,
+                                 final Link link,
+                                 final long pageSize,
+                                 final Map<String, Object> params) {
+
         try {
-            final Object streamId =  params.get(STREAM_ID);
-            final PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_PAGE);
+            final Object streamId = params.get(STREAM_ID);
+            if (offset == 0L && link.equals(HEAD)) {
+                return head(pageSize, streamId);
+            }
+            if (offset == 0L && link.equals(LAST)) {
+                return last(pageSize, streamId);
+            }
+            if (link.equals(PREVIOUS)) {
+                return forward(offset, pageSize, streamId);
+            }
+            if (link.equals(NEXT)) {
+                return backward(offset, pageSize, streamId);
+            }
+            return Stream.empty();
+        } catch (SQLException e) {
+            throw new JdbcRepositoryException(READING_STREAM_EXCEPTION, e);
+        }
+    }
+
+    private Stream<Event> last(final long pageSize, final Object streamId) throws SQLException {
+        final PreparedStatementWrapper ps;
+        ps = preparedStatementWrapperOf(LAST_RECORDS);
+        ps.setObject(1, streamId.toString());
+        ps.setLong(2, pageSize);
+        return streamOf(ps);
+    }
+
+    private Stream<Event> backward(final long offset, final long pageSize, Object streamId)
```

Comment:
final streamId - Object? or is it a UUID? - or can we set it to String?

Reply:
changed to Set String